processor/src/operations/field_ops.rs

```diff
@@ -162,6 +162,49 @@ impl Process {
         self.stack.copy_state(1);
         Ok(())
     }
+
+    /// Computes a single turn of exp accumulation for the given inputs. The stack is arranged
+    /// as follows (from the top):
+    /// - exponent of base for this turn - 1 element
+    /// - accumulated power of base so far - 1 element
+    /// - number which needs to be shifted to the right - 1 element
```

Comment:
I'm wondering if instead of using a helper register, we should put the bit value onto the stack itself. We already have several operation which need to assert that the first value on the stack is binary - and this could be another such operation. The initial layout of the stack could be: `[0, exp, acc, b, ...]`, and after executing `EXPAACC` it could be `[bit, (exp - 1) * bit + 1, b >> 1, ...]`.

This will probably increase cycle count (by 2 cycles?) - so, I'm not sure it's worth it. What do you think?

Reply:
We have a couple of operations in field operations where we have this constraint on the first being binary or not eg. `NOT`, `AND`, `OR`. At the moment they all are separate constraints and I think we can club all of them into one. 

I don't think this will impact much given expectedly the cycle count of `exp` is ~40 cycles. The Extra 2 cycles won't make a dent in the performance marginally. But in the case of a separate `pow2` operation, it will have an impact, which should be taking ~15 cycles to run, and this extra 2 will make it 17 cycles.

Reply:
I think we also have `CSWAP` and `CSWAPW` which require the same constraint. We could probably create a composite flag for all these operations to enforce that the first stack item is binary, and put the actual constraint into general constraints section.

Reply:
I was talking about operations in field operation module earlier. Although- it does makes sense to have all of them clubbed together as one constraint in general constraints.